src/main/java/net/logstash/logback/composite/CompositeJsonFormatter.java

```diff
@@ -119,40 +138,98 @@ public boolean isStarted() {
         return started;
     }
 
+    
     /**
-     * Create a reusable {@link JsonFormatter} bound to the given {@link OutputStream}.
+     * Write an event in the given output stream.
      * 
-     * @param outputStream the output stream used by the {@link JsonFormatter}
-     * @return {@link JsonFormatter} writing JSON content in the output stream
-     * @throws IOException thrown when unable to write in the output stream or when Jackson fails to produce JSON content
+     * @param event the event to write
+     * @param outputStream the output stream to write the event into
+     * @throws IOException thrown upon failure to write the event
      */
-    public JsonFormatter createJsonFormatter(OutputStream outputStream) throws IOException {
+    public void writeEvent(Event event, OutputStream outputStream) throws IOException {
+        Objects.requireNonNull(outputStream);
         if (!isStarted()) {
             throw new IllegalStateException("Formatter is not started");
         }
         
-        JsonGenerator generator = createGenerator(outputStream);
-        return new JsonFormatter(generator);
+        try (JsonFormatter formatter = this.pool.acquire()) {
+            formatter.writeEvent(outputStream, event);
+        }
     }
     
     
-    public class JsonFormatter implements Closeable {
+    /**
+     * Create a reusable {@link JsonFormatter} bound to a {@link DisconnectedOutputStream}.
+     * 
+     * @param stream the output stream used by the {@link JsonFormatter}
```

Comment:
remove the @param   (sorry, it was in my previous suggestion, but I didn't explicitly call it out)

Reply:
Indeed... but weird... I expected the javadoc compiler to complain about this (it does if the @param is missing tho).